client/me/profile/index.jsx

```diff
@@ -129,32 +128,19 @@ class Profile extends Component {
 							} ) }
 						>
 							<ToggleControl
-								disabled={ this.props.isFetchingUserSettings }
+								disabled={ this.props.isUpdatingUserSettings || this.props.isFetchingUserSettings }
 								checked={ this.props.getSetting( 'gravatar_profile_hidden' ) }
 								onChange={ this.toggleGravatarHidden }
 								label={ <UpdatedGravatarString gravatarProfileLink={ gravatarProfileLink } /> }
 							/>
 						</FormFieldset>
 
-						<FormFieldset
-							className={ classnames( {
-								'profile__is_dev_account-fieldset-is-loading': this.props.isFetchingUserSettings,
-							} ) }
-						>
-							<ToggleControl
-								disabled={ this.props.isFetchingUserSettings }
-								checked={ this.props.getSetting( 'is_dev_account' ) }
-								onChange={ this.toggleIsDevAccount }
-								label={ getIAmDeveloperCopy( this.props.translate ) }
-							/>
-						</FormFieldset>
-
 						<p className="profile__submit-button-wrapper">
 							<FormButton
 								disabled={ ! this.props.hasUnsavedUserSettings || this.props.getDisabledState() }
 								onClick={ this.getClickHandler( 'Save Profile Details Button' ) }
 							>
-								{ this.props.getDisabledState()
+								{ this.props.isUpdatingUserSettings
```

Comment:
In the whole component we are using `getDisabledState`, so I would keep it as is, to have consistency.

Also here https://github.com/Automattic/wp-calypso/pull/86558/files#diff-0faf1598511ba2e82b50d7d563dd2a58c54e6a1490e885a52151e2813a3a5e61R131

Reply:
Thanks for spotting! Hinted by the function name, I assumed `getDisabledState()` does more than just returning the `isUpdatingUserSettings` prop. A form can be considered disabled when data is still being fetched and when data is being updated. I guess all the function only accounts for when data is being updated :p I've updated the code as per your reco.